service/src/service.rs

```diff
@@ -213,6 +216,11 @@ impl crate::client::PostService for PostService {
     fn get_metadata(&self) -> &PostMetadata {
         &self.metadata
     }
+
+    fn interrupt_proof(&self) -> eyre::Result<()> {
+        self.stop.clone().store(true, Ordering::Relaxed);
```

Comment:
The Arc doesn't need to be cloned here
```suggestion
        self.stop.store(true, Ordering::Relaxed);
```